### PR TITLE
Make isVisible optionally inheritable

### PR DESCRIPTION
**Pull request by RaananW**
KHR Visibility, a new glTF extension, sets visibility as inherited value from a node's parent(s). Meaning, a node is only visible if it is set to be visible and all of its parents are visible. This is not what we are doing in Babylon. 

This PR adds a flag that should only be turned on when using the KHR visibility extension for each mesh of the loaded glTF. Enabling the flag will allow checking  mesh's visibility, based on its parents visibility as well.

This PR is a suggestion only, but it will be needed to support the latest extension. I understand there is a slight perf hit for the default behavior. I did my best to make sure we return early for the current behavior, without iteration the parents.

Test PG - #ABDDD6#118 (emulating enabling KHR interactivity).

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Meshes/abstractMesh.ts [hunk 1]
```diff
@@ -536,10 +536,35 @@ export abstract class AbstractMesh extends TransformNode implements IDisposable,
      */
     public alphaIndex = Number.MAX_VALUE;
 
+    /**
+     * If set to true, a mesh will only be visible only if its parent(s) are also visible (default is false)
+     */
+    public inheritedVisibility = false;
```
Comment (deltakosh) [inline packages/dev/core/src/Meshes/abstractMesh.ts:542]: Can you stock it on our property bag to avoid increasing the number of class members?

Comment (RaananW) [reply packages/dev/core/src/Meshes/abstractMesh.ts:542]: of course. let me do that now

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15275/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15275/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15275/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15275/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15275/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15275/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15275/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15275/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15275/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15275/merge/testResults/webgpuplaywright/index.html

File: packages/dev/core/src/Meshes/abstractMesh.ts [hunk 2]
```diff
@@ -231,6 +231,12 @@ class _InternalAbstractMeshDataInfo {
      * We use that as a clue to force the material to sideOrientation = null
      */
     public _sideOrientationHint = false;
+
+    /**
+     * @internal
+     * if this is set to true, the mesh will be visible only if its parent(s) are also visible
+     */
+    public _inheritedVisibility = false;
```
Comment (deltakosh) [inline packages/dev/core/src/Meshes/abstractMesh.ts:239]: sorry I was unclear: I meant to move it as part of  _internalAbstractMeshDataInfo

File: packages/dev/core/src/Meshes/abstractMesh.ts [hunk 3]
```diff
@@ -536,10 +542,41 @@ export abstract class AbstractMesh extends TransformNode implements IDisposable,
...
+    public get isVisible(): boolean {
+        if (!this._isVisible || !this.inheritedVisibility || !this._parentNode) {
+            return this._isVisible;
+        }
+        if (this._isVisible) {
+            let parent: Nullable<Node> = this._parentNode;
+            while (parent) {
+                const parentVisible = (parent as AbstractMesh).isVisible;
+                if (typeof parentVisible !== "undefined") {
+                    return this._isVisible && parentVisible;
```
Comment (AmoebaChant) [inline packages/dev/core/src/Meshes/abstractMesh.ts:569]: This checks _isVisible but it's in a block that you'll only reach if _isVisible is true - can this just be return parentVisible?

File: packages/dev/core/src/Meshes/abstractMesh.ts [hunk 4]
```diff
@@ -536,10 +542,41 @@ export abstract class AbstractMesh extends TransformNode implements IDisposable,
...
     public alphaIndex = Number.MAX_VALUE;
 
+    /**
+     * If set to true, a mesh will only be visible only if its parent(s) are also visible (default is false)
+     */
+    public get inheritedVisibility(): boolean {
+        return this._internalAbstractMeshDataInfo._inheritedVisibility;
+    }
+
+    public set inheritedVisibility(value: boolean) {
```
Comment (AmoebaChant) [inline packages/dev/core/src/Meshes/abstractMesh.ts:552]: Naming feedback: I may think that inheritedVisibility is the visibility value that was inherited, not that it should inherit visibility or not.  Would inheritVisibility be more clear?

**Review by deltakosh — APPROVED** [review]

File: packages/dev/core/src/Meshes/abstractMesh.ts (hunk 3 above)
Comment (RaananW) [reply packages/dev/core/src/Meshes/abstractMesh.ts:569]: makes sense :-)

File: packages/dev/core/src/Meshes/abstractMesh.ts (hunk 4 above)
Comment (RaananW) [reply packages/dev/core/src/Meshes/abstractMesh.ts]: Done! thanks for the suggestion :-)

**Pull request merged**